azext_iot/deviceupdate/_help.py

```diff
@@ -260,3 +260,537 @@ def load_deviceupdate_help():
+          device is connected to the hub and reports its properties.
+    """
+
+    helps["iot device-update device group list"] = """
+        type: command
+        short-summary: List device groups within an instance.
+
+        examples:
+        - name: List device groups.
+          text: >
+            az iot device-update device group list -n {account_name} -i {instance_name}
+
+        - name: List device groups in a desired order, in this case by deviceCount.
+          text: >
+            az iot device-update device group list -n {account_name} -i {instance_name} --order-by deviceCount
+    """
+
+    helps["iot device-update device group show"] = """
+        type: command
+        short-summary: Show details about a device group including the best update and update compliance.
+
+        examples:
+        - name: Show a device group.
+          text: >
+            az iot device-update device group show -n {account_name} -i {instance_name} --group-id {device_class_id}
+
+        - name: Show the best updates available for a device group. This flag modifies the command to returns a list.
+          text: >
+            az iot device-update device group show -n {account_name} -i {instance_name} --group-id {device_group_id} --best-updates
+
+        - name: Show update compliance for a device group.
+          text: >
+            az iot device-update device group show -n {account_name} -i {instance_name} --group-id {device_group_id} --update-compliance
+    """
+
+    helps["iot device-update device group delete"] = """
+        type: command
+        short-summary: Delete a device group.
+        long-summary: >
+          Device groups are not automatically cleaned up but are retained for history purposes. This operation
+          can be used if there is no need for the group or need to retain history for it. If a device is ever connected
+          again for a group after the group was deleted it will be automatically re-created with no history.
+
+        examples:
+        - name: Delete a device group.
+          text: >
+            az iot device-update device group delete -n {account_name} -i {instance_name} --group-id {device_class_id}
+
+        - name: Delete a device group skipping the confirmation prompt.
+          text: >
+            az iot device-update device group delete -n {account_name} -i {instance_name} --group-id {device_group_id} -y
+    """
+
+    helps["iot device-update device deployment"] = """
+        type: group
+        short-summary: Device deployment management.
+        long-summary: Deployments will apply a desired compatible update against a target device group distributing the update
+          across device classes within the group. Cloud-initiated rollback policy can be optionally configured.
+    """
+
+    helps["iot device-update device deployment create"] = """
```

Comment:
Is there benefit in adding a command to create deployments for all device groups belonging to a device class (say a user wants to do a global deployment)? With current implementation, the user will have to run this command in a loop for each group under the device class when they want to create a deployment for all device groups.

Reply:
Device groups are meant for logically separating or organizing a set of devices and device classes should always belong to a group and are generated based on compatibility specifications (also from a API organization perspective it is interesting classes can be addressed at the top level or within a group). When you deploy to a group (only thing you can do today) all classes within the group get deployed to in separate deployments automatically.

Good thought on a global deployment, but it is unclear how common of a use case that would be, because if you have a bunch of devices that can support the same update compatibility wise, why not group them together and deploy to the group (which supports the scale)?

I also think for the service, we should observe some customer experience/insights to help gauge deployment productivity boosters.

Reply:
Makes sense, looking at usage metrics will allow us to understand the areas where we can provide useful features/functionality.